zaza/openstack/charm_tests/trilio/setup.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Code for configuring Trilio."""
+
+import logging
+import os
+
+import zaza.model as zaza_model
+import zaza.openstack.utilities.juju as juju_utils
+
+
+def basic_setup():
+    """Run setup for testing Trilio.
+
+    Setup for testing Trilio is currently part of functional
+    tests.
+    """
+    logging.info("Configuring NFS Server")
+    nfs_server_ip = zaza_model.get_app_ips("nfs-server-test-fixture")[0]
+    trilio_wlm_unit = zaza_model.get_first_unit_name("trilio-wlm")
+
+    nfs_shares_conf = {"nfs-shares": "{}:/srv/testing".format(nfs_server_ip)}
+    _trilio_services = ["trilio-wlm", "trilio-data-mover"]
+    for juju_service in _trilio_services:
+        zaza_model.set_application_config(juju_service, nfs_shares_conf)
```

Comment:
In general doing this may be tricky as you don't know what the model will look like before you or someone else executes this code. If the config value happens to be the same as you are trying to set the test code will execute indefinitely waiting for something to happen that won't.

Having said that since you are setting a value from the underlay that most likely will be different every time it will most likely be ok here.

The [config_change](https://github.com/openstack-charmers/zaza-openstack-tests/blob/719c4bfacf7b6d662e21452195e576d136484963/zaza/openstack/charm_tests/test_utils.py#L177) context manager can be useful for these kinds of changes as it checks the already set value for you.